docs/guide/common_addons.md

```diff
@@ -71,4 +71,155 @@ dashboard accessible from the outside via `https` on port `443`.
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: registry-storage
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+```
+
+### Run the registry
+Now you can run the Docker registry:
+
+```
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: kube-registry-v0
+  labels:
+    k8s-app: kube-registry-upstream
+    version: v0
+    kubernetes.io/cluster-service: "true"
+spec:
+  replicas: 1
+  selector:
+    k8s-app: kube-registry-upstream
+    version: v0
+  template:
+    metadata:
+      labels:
+        k8s-app: kube-registry-upstream
+        version: v0
+        kubernetes.io/cluster-service: "true"
+    spec:
+      containers:
+      - name: registry
+        image: registry:2.5.1
+        resources:
+          limits:
+            cpu: 100m
+            memory: 100Mi
+          requests:
+            cpu: 100m
+            memory: 100Mi
+        env:
+        - name: REGISTRY_HTTP_ADDR
+          value: :5000
+        - name: REGISTRY_STORAGE_FILESYSTEM_ROOTDIRECTORY
+          value: /var/lib/registry
+        volumeMounts:
+        - name: image-store
+          mountPath: /var/lib/registry
+        ports:
+        - containerPort: 5000
+          name: registry
+          protocol: TCP
+      volumes:
+      - name: image-store
+        persistentVolumeClaim: 
+          claimName: registry-storage
+```
+
+### Expose registry in the cluster
+```
+apiVersion: v1
+kind: Service
+metadata:
+  name: kube-registry
+  labels:
+    k8s-app: kube-registry-upstream
+    kubernetes.io/cluster-service: "true"
+    kubernetes.io/name: "KubeRegistry"
+spec:
+  selector:
+    k8s-app: kube-registry-upstream
+  ports:
+  - name: registry
+    port: 5000
+    protocol: TCP
+```
+
+### Expose the registry on each node
+Now that there is a running Service, you need to expose it onto each Kubernetes Node so that Docker will see it as localhost.
+
+```
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: registry-proxy-v0
+  labels:
+    k8s-app: kube-registry-proxy
+    kubernetes.io/cluster-service: "true"
+    version: v0.4
+spec:
+  template:
+    metadata:
+      labels:
+        k8s-app: kube-registry-proxy
+        kubernetes.io/name: "kube-registry-proxy"
+        kubernetes.io/cluster-service: "true"
+        version: v0.4
+    spec:
+      containers:
+      - name: kube-registry-proxy
+        image: gcr.io/google_containers/kube-registry-proxy:0.4
+        resources:
+          limits:
+            cpu: 100m
+            memory: 50Mi
+        env:
+        - name: REGISTRY_HOST
+          value: kube-registry
+        - name: REGISTRY_PORT
+          value: "5000"
+        ports:
+        - name: registry
+          containerPort: 80
+          hostPort: 5000
+```
+
+### Access registry from outside
+Through a ssh tunnel you can push or pull images from your cluster registry. At first, export your local ip: 
+```
+export local_ip=$(ipconfig getifaddr en0)
```

Comment:
is this working? i thought a tag named 10.46.0.1:5000 in docker is not the same as localhost:5000 is not the same as 127.0.0.1:5000

Reply:
The tunnel works fine whereas push/pull via. port-forwarding on Mac doesn't work as expected. Other user are facing the same issue. It seems like forwarded port is not being used by docker for Mac. `Error response from daemon: Get http://localhost:5000/v2/: dial tcp [::1]:5000: getsockopt: connection refused`. The annoying aspect by using the tunnel is to change the local_ip in "insecure repos" on the local machine after getting a new one.